lib/redis_client.rb

```diff
@@ -717,6 +737,7 @@ def ensure_connected(retryable: true)
       preferred_error = nil
       begin
         connection = raw_connection
+        connection.retry_attempt = config.retriable?(tries) ? tries : nil
```

Comment:
Would it be correct to set `connection.retry_attempt = nil` below in the non-retryable branch after we have the connection? 

https://github.com/redis-rb/redis-client/blob/8620e74207696973365f066baeac10126e20ad60/lib/redis_client.rb#L767